File: aurea/src/main/java/com/emdtech/aurea/dto/ProductResponse.java
package com.emdtech.aurea.dto;

import java.time.LocalDateTime;
import java.util.List;

public record ProductResponse(

        Long id,

        Long categoryId,
        String categoryName,

        String name,
        String description,

        Boolean active,

        List<ProductPriceResponse> prices,

        LocalDateTime createdAt,
        LocalDateTime updatedAt

) {
}

File: aurea-angular/src/app/core/models/product-response.ts
import { ProductPriceResponse } from './product-price-response';

export interface ProductResponse {
  id: number;
  categoryId: number;
  categoryName: string;
  name: string;
  description: string | null;
  active: boolean;
  prices: ProductPriceResponse[];
  createdAt: string;
  updatedAt: string;
}
